addon/components/file-upload.ts

```diff
@@ -0,0 +1,125 @@
+import Component from '@glimmer/component';
+import { inject as service } from '@ember/service';
+import { action } from '@ember/object';
+
+import uuid from '../system/uuid';
+import File from '../file';
+import Queue from '../queue';
+import FileQueueService, { DEFAULT_QUEUE } from '../services/file-queue';
+
+interface FileUploadArgs {
+  name?: string;
```

Comment:
I think we should drop or at least deprecate this in favour of `queue`.

Reply:
Yeah, deprecation warnings as they are still part of the public API. A contract for the addon to stick to until v6 then.